econml/metalearners.py

```diff
@@ -150,39 +149,42 @@ def fit(self, Y, T, X, inference=None):
         self : an instance of self.
         """
         # Check inputs
+        if X is None:
+            X = np.ones((Y.shape[0], 1))
         Y, T, X, _ = check_inputs(Y, T, X, multi_output_T=False)
         T = self._label_encoder.fit_transform(T)
         T = self._one_hot_encoder.fit_transform(T.reshape(-1, 1))
-        self._d_t = T.shape[1:]
-        self._d_y = Y.shape[1:]
+        self._d_t = T.shape[1] - 1
```

Comment:
To be consistent with other classes, this should probably always be a 1-tuple rather than a raw value.  That is,
```suggestion
        self._d_t = T.shape[1] - 1,
```